components/test_util/src/lib.rs

```diff
@@ -93,3 +93,50 @@ pub fn alloc_port() -> u16 {
         }
     }
 }
+
+static MEM_DISK: &str = "TIKV_TEST_MEMORY_DISK_MOUNT_POINT";
+
+/// Gets a temporary path in memory.
+///
+/// The returned path will point to memory only when memory disk is available
+/// and specified.
+pub fn temp_dir_in_mem(prefix: impl Into<Option<&'static str>>) -> tempfile::TempDir {
+    let mut builder = tempfile::Builder::new();
+    if let Some(prefix) = prefix.into() {
+        builder.prefix(prefix);
+    }
+    if let Ok(dir) = env::var(MEM_DISK) {
+        debug!("using memory disk"; "path" => %dir);
+        builder.tempdir_in(dir).unwrap()
+    } else {
+        builder.tempdir().unwrap()
+    }
+}
+
+pub struct GuaranteeOnPhysicalDisk {
+    mem_disk: Option<String>,
+}
+
+impl Drop for GuaranteeOnPhysicalDisk {
+    fn drop(&mut self) {
+        if let Some(p) = self.mem_disk.take() {
+            env::set_var(MEM_DISK, p);
+        }
+    }
+}
+
+/// Memory disk is provided for CI. By default, all raftstore test cases
+/// are run on memory disk. But in some cases, testing IO stats from system
+/// for example, you may want to run cases on physical disk instead.
+///
+/// The correctness of this API depends on running test case one by one.
+pub fn guarantee_on_physical_disk() -> GuaranteeOnPhysicalDisk {
```

Comment:
Could we add some config (like `try_men_disk`) to control using what kind of disk instead? And setting the default value based on `#[cfg(any(test))]`, if `try_men_disk` is true then call `temp_dir_in_mem` in `test_raftstore`.